File: db4oj.tests/src/com/db4o/db4ounit/common/assorted/SimplestPossibleNullMemberTestCase.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.assorted;

import db4ounit.*;
import db4ounit.extensions.*;


public class SimplestPossibleNullMemberTestCase extends AbstractDb4oTestCase{

    public static class Item{
        public Item _item;
    }
    
    protected void store() throws Exception {
        store(new Item());
    }
    
    public void test(){
        Item item = (Item) retrieveOnlyInstance(Item.class);
        Assert.isNull(item._item);
    }

}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Assorted/SimplestPossibleNullMemberTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Tests.Common.Assorted;

namespace Db4objects.Db4o.Tests.Common.Assorted
{
	public class SimplestPossibleNullMemberTestCase : AbstractDb4oTestCase
	{
		public class Item
		{
			public SimplestPossibleNullMemberTestCase.Item _item;
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Store(new SimplestPossibleNullMemberTestCase.Item());
		}

		public virtual void Test()
		{
			SimplestPossibleNullMemberTestCase.Item item = (SimplestPossibleNullMemberTestCase.Item
				)((SimplestPossibleNullMemberTestCase.Item)RetrieveOnlyInstance(typeof(SimplestPossibleNullMemberTestCase.Item
				)));
			Assert.IsNull(item._item);
		}
	}
}
